lib/token-persistence.spec.js

```diff
@@ -27,67 +16,63 @@ describe('Token persistence', function() {
-    it('creates it with an empty array', async function() {
+    it('does nothing', async function() {
       await persistence.initialize()
-      const json = JSON.parse(await readFile(path))
+      expect(githubAuth.getAllTokenIds()).to.deep.equal([])
+    })
 
-      expect(json).to.deep.equal([])
+    it('saving creates an empty file', async function() {
+      await persistence.initialize()
+
+      await persistence.save()
+
+      const json = JSON.parse(await readFile(path))
+      expect(json).to.deep.deep.equal([])
     })
   })
 
   context('when the file exists', function() {
-    it('loads the contents', async function() {
-      const initialTokens = ['a', 'b', 'c'].map(char => char.repeat(40))
+    const initialTokens = ['a', 'b', 'c'].map(char => char.repeat(40))
+
+    beforeEach(async function() {
       fs.writeFileSync(path, JSON.stringify(initialTokens))
+    })
 
+    it('loads the contents', async function() {
       await persistence.initialize()
-
       expect(githubAuth.getAllTokenIds()).to.deep.equal(initialTokens)
     })
-  })
-
-  context('when shutting down', function() {
-    it('writes added tokens to the file', async function() {
-      const initialTokens = ['a', 'b', 'c'].map(char => char.repeat(40))
-      fs.writeFileSync(path, JSON.stringify(initialTokens))
 
-      const newToken = 'e'.repeat(40)
-      const expected = initialTokens.slice()
-      expected.push(newToken)
+    context('when tokens are added', function() {
+      it('writes them to the file', async function() {
+        const newToken = 'e'.repeat(40)
+        const expected = initialTokens.slice()
+        expected.push(newToken)
 
-      await persistence.initialize()
-      githubAuth.addGithubToken(newToken)
-      await persistence.stop()
+        await persistence.initialize()
+        githubAuth.addGithubToken(newToken)
+        await persistence.noteTokenAdded(newToken)
 
-      const json = JSON.parse(await readFile(path))
-      expect(json).to.deep.equal(expected)
+        const json = JSON.parse(await readFile(path))
+        expect(json).to.deep.equal(expected)
+      })
     })
-  })
 
-  context('time has elapsed', function() {
-    it('writes added tokens to the file', async function() {
-      const addedTokens = ['d', 'e'].map(char => char.repeat(40))
+    context('when tokens are removed', function() {
+      it('writes them to the file', async function() {
+        const expected = Array.from(initialTokens)
+        const toRemove = expected.pop()
 
-      await persistence.initialize()
-
-      addedTokens.forEach(githubAuth.addGithubToken)
+        await persistence.initialize()
 
-      // Fake time passing to trigger autosaving, then give the save a brief
-      // moment of real time to complete before reading.
-      clock.tick(5000)
-      clock.restore()
-      await sleep(200)
+        githubAuth.rmGithubToken(toRemove)
+        await persistence.noteTokenAdded(toRemove)
```

Comment:
Shouldn't this be `noteTokenRemoved`?

Reply:
Ah thanks, I caught that in #1939 but forgot to fix it here.